### PR TITLE
Test warnings when handler doesnt filter resources

### PR DESCRIPTION
**Pull request by theninj4**
I've added to the test helper so every request goes through some basic pagination validation.

The goal is to use the pagination block to work out how many resources the data handler *thinks* should be present in the response, then compare that with however many resources appear in the resulting payload. If the two differ, the handler isn't properly implementing filtering.

Right now the in-memory-handler used by the test suite for this project doesn't implement filtering (we rely on the post processing to clean it up). As such, running the test suit should show a whole bunch of warnings:
```
  Testing jsonapi-server
    Searching for resources
      applying filter
        ✓ value of wrong type should error
!!!!!!!!!!!!
http://localhost:16006/rest/articles?filter[title]=How%20to%20AWS
WARNING: Pagination count doesn't match resource count.
This usually indicates the resource hanlder is not filtering correctly!
!!!!!!!!!!!!
        ✓ equality for strings
!!!!!!!!!!!!
http://localhost:16006/rest/articles?filter[views]=10
WARNING: Pagination count doesn't match resource count.
This usually indicates the resource hanlder is not filtering correctly!
!!!!!!!!!!!!
```

Within this project, those warnings are fine and expected. Nobody should ever be using the in-memory-handler, it simply needs to validate the rest of the project functions correctly. If any handlers use this test suite to validate their functionality (which is surprisingly effective at proving all the functionality is covered) then seeing these warnings is bad and it means something's broken.

### CODE REVIEW

**Comment by theninj4** [comment]
Helps #161

File: test/helpers.js [hunk 1]
```diff
@@ -56,9 +56,33 @@ testHelpers.validateJson = function(json) {
   assert.ok(json.links instanceof Object, "Response should have a links block");
   assert.ok(!(json.errors instanceof Object), "Response should not have any errors");
   assert.equal(typeof json.links.self, "string", "Response should have a \"self\" link");
+  testHelpers.validatePagination(json);
   return json;
 };
 
+testHelpers.validatePagination = function(json) {
+  if (!json.meta.page) return;
+  if (!(json.data instanceof Array)) return;
```
Comment (pmcnr-hx) [inline test/helpers.js]: `Array.isArray` seems to be recommend when supported (as in Node's case) since it covers some corner cases that `instanceof Array` doesn't.

File: test/helpers.js [hunk 2]
```diff
@@ -56,9 +56,33 @@ testHelpers.validateJson = function(json) {
...
+
+  var page = json.meta.page;
+  var expectedCount = null;
+  if ((page.offset + page.limit) > page.total) {
```
Comment (pmcnr-hx) [inline test/helpers.js]: I personally wouldn't bother with the redundant `()`. Do you reckon it deters from readability?

**Comment by pmcnr-hx** [comment]
Just minor non-blocker comments which I will leave at your discretion: 👍 :shipit:

File: test/helpers.js (hunk 1 above)
Comment (theninj4) [inline test/helpers.js]: The rest of this project uses `instanceof Array`, so we should stick with it.

File: test/helpers.js (hunk 2 above)
Comment (theninj4) [inline test/helpers.js]: I see the brackets as an important part of identifying the left hand side from the right hand side.
